examples/plot_knockoff_aggregation.py

```diff
@@ -37,142 +38,146 @@
 from hidimstat.statistical_tools.multiple_testing import fdp_power
 from hidimstat._utils.scenario import multivariate_1D_simulation_AR
 
-plt.rcParams.update({"font.size": 26})
+plt.rcParams.update({"font.size": 12})
 
 # Number of observations
-n_subjects = 500
-# Number of variables
-n_clusters = 500
-# Correlation parameter
-rho = 0.7
-# Ratio of number of variables with non-zero coefficients over total
-# coefficients
-sparsity = 0.1
-# Desired controlled False Discovery Rate (FDR) level
-fdr = 0.1
-seed = 45
-n_bootstraps = 25
-n_jobs = 10
-runs = 20
-
-rng = check_random_state(seed)
-seed_list = rng.randint(1, np.iinfo(np.int32).max, runs)
-
-
-def single_run(
-    n_subjects, n_clusters, rho, sparsity, fdr, n_bootstraps, n_jobs, seed=None
-):
-    # Generate data
-    X, y, _, non_zero_index = multivariate_1D_simulation_AR(
-        n_subjects, n_clusters, rho=rho, sparsity=sparsity, seed=seed
-    )
-
-    # Use model-X Knockoffs [1]
-    selected, test_scores, threshold, X_tildes = model_x_knockoff(
-        X,
-        y,
-        estimator=LassoCV(
-            n_jobs=n_jobs,
-            verbose=0,
-            max_iter=1000,
-            cv=KFold(n_splits=5, shuffle=True, random_state=0),
-            tol=1e-6,
-        ),
-        n_bootstraps=1,
-        random_state=seed,
-    )
-    mx_selection, _ = model_x_knockoff_pvalue(test_scores, fdr=fdr)
-    fdp_mx, power_mx = fdp_power(mx_selection, non_zero_index)
-
-    # Use p-values aggregation [2]
-    selected, test_scores, threshold, X_tildes = model_x_knockoff(
-        X,
-        y,
-        estimator=LassoCV(
-            n_jobs=n_jobs,
-            verbose=0,
-            max_iter=1000,
-            cv=KFold(n_splits=5, shuffle=True, random_state=0),
-            tol=1e-6,
-        ),
-        n_bootstraps=n_bootstraps,
-        n_jobs=n_jobs,
-        random_state=seed,
-    )
-    aggregated_ko_selection, _, _ = model_x_knockoff_bootstrap_quantile(
-        test_scores, fdr=fdr, gamma=0.3
-    )
-
-    fdp_pval, power_pval = fdp_power(aggregated_ko_selection, non_zero_index)
+n_subjects = 100
+
+
+def different_result(n_subjects):
+    # Number of variables
+    n_clusters = 100
+    # Correlation parameter
+    rho = 0.3
+    # Ratio of number of variables with non-zero coefficients over total
+    # coefficients
+    sparsity = 0.1
+    # Desired controlled False Discovery Rate (FDR) level
+    fdr = 0.1
+    snr = 10
+    seed = 45
+    n_bootstraps = 25
+    runs = 20
+    n_jobs = None
+    joblib_verbose = 0
+
+    rng = check_random_state(seed)
+    seed_list = rng.randint(1, np.iinfo(np.int32).max, runs)
+
+    def single_run(n_subjects, n_clusters, rho, sparsity, fdr, n_bootstraps, seed=None):
```

Comment:
We should avoid creating functions in examples because this adds complexity. But sometimes this is not avoidable. In any case, we should avoid nested functions even more !